cunit/message.testc

```diff
@@ -1391,4 +1391,17 @@ static void test_parse_bogus_charset_params(void)
 #undef TESTCASE
 }
 
+/*
+ * Verifies that message_parse_received_date() does not read
+ * uninitialized data in the second call to message_parse_string()
+ */
+static void test_parse_received_semicolon(void)
+{
+   static const char msg[] = "Received: abc;\r\n\r\nd";
+   struct body body;
+   memset(&body, 0x45, sizeof(body));
+   CU_ASSERT_EQUAL(message_parse_mapped(msg, sizeof(msg)-1, &body, NULL), 0);
+   CU_ASSERT_STRING_EQUAL(body.received_date, "");
```

Comment:
The test fails here with the fix, because when a semicolon with nothing after it is found, the original header value is used as the date string, which in this case is "abc;".  But this line expects the empty string.  It should expect "abc;", like this:  

```suggestion
   CU_ASSERT_STRING_EQUAL(body.received_date, "abc;");
```

Reply:
Neither `"abc"` nor `""` looks right to me. A non-existent date-time part in the header value should result in a NULL received_date value. The code building on body->received_date in lmtpd and jmap_mail is already prepared to handle NULL received_dates.